test/middleware/CompanyAuthMiddleware.test.ts

```diff
@@ -0,0 +1,184 @@
+import 'reflect-metadata';
+
+import { Arg, Substitute} from '@fluffy-spoon/substitute';
+import { Session, SessionStore } from 'ch-node-session-handler';
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { ISignInInfo } from 'ch-node-session-handler/lib/session/model/SessionInterfaces';
+import { expect } from 'chai';
+import { NextFunction, Request, Response } from 'express';
+
+import 'app/controllers/index';
+import { CompanyAuthMiddleware } from 'app/middleware/CompanyAuthMiddleware';
+import { Appeal } from 'app/models/Appeal';
+import { ApplicationData, APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { CompanyAuthConfig } from 'app/models/CompanyAuthConfig';
+import { JwtEncryptionService } from 'app/modules/jwt-encryption-service/JwtEncryptionService';
+import { SESSION_NOT_FOUND_ERROR } from 'app/utils/CommonErrors';
+
+import { createSubstituteOf } from 'test/SubstituteFactory';
+
+describe('Company Authentication Middleware', () => {
+
+    const appeal: Appeal = {
+        penaltyIdentifier: {
+            companyNumber: 'SC123123',
+            userInputPenaltyReference: 'A1231234',
+            penaltyReference: 'A1231234'
+        },
+        reasons: {
+            other: {
+                title: '',
+                description: ''
+            }
+        }
+    };
+
+    const companyAuthConfig: CompanyAuthConfig = {
+        oathScopePrefix: 'MOCK',
+        accountUrl: 'MOCK',
+        accountRequestKey: 'MOCK',
+        accountClientId: 'MOCK',
+        chsUrl: 'MOCK',
+    };
+
+    const sessionStoreForAuthConfig = {
+        sessionCookieName: '__SID',
+        sessionCookieDomain: 'rebel1.aws.chdev.org',
+        sessionCookieSecureFlag: 'true',
+        sessionTimeToLiveInSeconds: 3600
+    };
+
+    const redirectUrl: string = 'MOCK/oauth2/authorise?client_id=MOCK&redirect_uri=MOCK/oauth2/user/callback&response_type=code&scope=MOCKSC123123&state=';
```

Comment:
Because you use MOCL for all elements of configuration it won't detect cases where you implemented bad string concatenation and for example `accountUrl` is used twice while `chsUrl` is never used.

Provide unique sample values for tests that will give us a chance to detect potential code issue.